EIPS/eip-2027.md

```diff
@@ -0,0 +1,74 @@
+category: Core
+created: 2019-05-14
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+Ethereum starts counting the number of storage slots filled and emptied in the contracts. Since the number of pre-existing slots is not currently accounted
+in the state, effectively, only net change in the number of slots is tracked. In the subsequent change, called *Gross contract size accounting*, the total
+number of storage slots starts being tracked.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+This is part of the State Rent roadmap. This particular change introduced initial, net accounting of the number of the contract storage slots. Though not very
+useful on its own, it makes it possible to introduce gross accounting of the number of storage slots, which is useful for number of things:
+1. Gas cost of operations suchs as `SLOAD` and `SSTORE` will need to be increased to compensate for extra bandwidth consumed by the block proofs. Although in
+the beginning the cost would be fixed, it will later be automatically calibrated depending on the size of the contract `SLOAD` and `SSTORE` operate on.
+2. Snapshot sync protocols, like *fast sync*, *warp sync*, *firehose*, *red queen*, and perhaps others, will benefit from having the correct size of the
+contract storage present in the state (and therefore being provable via Merkle proofs).
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not track the number of contract storage slots at all, and producing such number given the downloaded state cannot be done in
+constant *O(1)* time.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Each contract gets a new uint64 field, called `storagesize`. On and after block C, the semantics of the operation `SSTORE` (`location`, `value`) changes as follows:
```

Comment:
Each contract or account? If contract, can you specify what constitutes as a contract (has non-empty code)?

Reply:
Added clarification, thank you